scripts/language_model/large_word_language_model.py

```diff
@@ -0,0 +1,334 @@
+"""
+Large Word Language Model
+===================
+
+This example shows how to build a word-level language model on Google Billion Words dataset
+with Gluon NLP Toolkit.
+By using the existing data pipeline tools and building blocks, the process is greatly simplified.
+
+We implement the LSTM 2048-512 language model proposed in the following work.
+
+@article{jozefowicz2016exploring,
+ title={Exploring the Limits of Language Modeling},
+ author={Jozefowicz, Rafal and Vinyals, Oriol and Schuster, Mike and Shazeer, Noam and Wu, Yonghui},
+ journal={arXiv preprint arXiv:1602.02410},
+ year={2016}
+}
+
+"""
+
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import time
+import math
+import os
+import sys
+import argparse
+import numpy as np
+import mxnet as mx
+from mxnet import gluon, autograd
+import gluonnlp as nlp
+from sampler import LogUniformSampler
+
+curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
+sys.path.append(os.path.join(curr_path, '..', '..'))
+
+###############################################################################
+# Arg parser
+###############################################################################
+parser = argparse.ArgumentParser(description=
+                                 'Gluon-NLP Big LSTM 2048-512 Language Model on GBW')
+parser.add_argument('--save', type=str, default='model.params',
+                    help='path to save the final model.')
+parser.add_argument('--emsize', type=int, default=512,
```

Comment:
The embedding size must equal to the nproj?

Reply:
Can you add an explanation on the embedding size not necessarily to be the same with the projection size in the argument description?

Reply:
Added now.

Reply:
There is a typo though, `embsize` vs `emsize`